src/index.ts

```diff
@@ -1,61 +1,105 @@
-import { Reshuffle, BaseConnector, EventConfiguration } from 'reshuffle-base-connector'
+import amqplib, { Connection, Channel, Options, ConsumeMessage } from 'amqplib'
+import { Reshuffle, EventConfiguration, BaseConnector } from 'reshuffle-base-connector'
 
-export interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  var1: string
-  // ...
+export interface AMQPConnectorConfigOptions {
+  queueUrl: string
+  queueName: string
+  queueOptions?: Options.AssertQueue
 }
 
-export interface _CONNECTOR_NAME_ConnectorEventOptions {
-  option1?: string
-  // ...
+export interface AMQPConnectorEventOptions {
+  consumeOptions?: Options.Consume
 }
 
-export default class _CONNECTOR_NAME_Connector extends BaseConnector<
-  _CONNECTOR_NAME_ConnectorConfigOptions,
-  _CONNECTOR_NAME_ConnectorEventOptions
-> {
-  // Your class variables
-  var1: string
+export default class AMQPConnector extends BaseConnector {
+  private client: typeof amqplib
+  private connectionPublisher?: Connection
+  private connectionConsumer?: Connection
+  private channelPublisher?: Channel
+  private channelConsumer?: Channel
+  private handlersByEventId: { [eventId: string]: any } = {}
 
-  constructor(app: Reshuffle, options?: _CONNECTOR_NAME_ConnectorConfigOptions, id?: string) {
+  constructor(app: Reshuffle, options: AMQPConnectorConfigOptions, id?: string) {
     super(app, options, id)
-    this.var1 = options?.var1 || 'initial value'
-    // ...
+    this.client = amqplib
+    if (!options) {
+      throw new Error('Empty connection config')
+    }
   }
 
-  onStart(): void {
-    // If you need to do something specific on start, otherwise remove this function
+  on(options: AMQPConnectorEventOptions, handler: any, eventId: string): EventConfiguration {
+    if (!eventId) {
+      eventId = `AMQP/${this.configOptions?.queueUrl}/${this.id}`
+    }
+
+    const event: EventConfiguration = new EventConfiguration(eventId, this, options)
+    this.eventConfigurations[eventId] = event
+    this.handlersByEventId[eventId] = handler
+    return event
   }
 
-  onStop(): void {
-    // If you need to do something specific on stop, otherwise remove this function
+  async onStart(): Promise<void> {
+    await this.createQueue(true)
+    await this.createQueue(false)
```

Comment:
Why are we creating both consumer and publisher automatically?
At least for consumer we can check whether `on` was called?